parser.go

```diff
@@ -52,7 +54,7 @@ func init() {
 		Seq(&rIgnore, &rIdentifier, &rLEFTARROW, &rExpression))
 
 	rExpression.Ope = Seq(&rSequence, Zom(Seq(&rSLASH, &rSequence)))
-	rSequence.Ope = Zom(&rPrefix)
+	rSequence.Ope = Oom(&rPrefix)
```

Comment:
@CWood1, I have faithfully ported Bryan Ford's code in 'Figure 1. PEG formally describing its own ASCII syntax' at http://bford.info/pub/lang/peg.pdf. Is there any reason to change from `Zom` to `Oom`?

```
# Hierarchical syntax
Grammar    <- Spacing Definition+ EndOfFile
Definition <- Identifier LEFTARROW Expression
Expression <- Sequence (SLASH Sequence)*
Sequence   <- Prefix*
Prefix     <- (AND / NOT)? Suffix
Suffix     <- Primary (QUESTION / STAR / PLUS)?
Primary    <- Identifier !LEFTARROW
            / OPEN Expression CLOSE
            / Literal / Class / DOT
```

Reply:
Your change looks better though.

Reply:
I've investigated a little deeper into this now. I made this change because a bunch of tests started failing, and this made them pass again. Turns out my mistake was to leave off a bounds check, which got added in again later.

I think I prefer the `Oom` to the `Zom`, but by that same token, I can't really think of any technical reason to not revert that line back to `Zom`, other than the fairly airy "surely people will need to match on something here?".

What are your thoughts? I'm quite happy to revert that line back to `Zom`, or to leave it as is, and can see arguments for either case, since both seem to yield meaningfully correct results in all cases.